src/unix/system.c

```diff
@@ -341,7 +345,25 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
+	    xdg_data_home_dir = malloc(sizeof(char) * PATH_MAX);
```

Comment:
You could also use some equivalent to `strlen(homedir) + 1 + strlen(".local/share")` instead of `PATH_MAX`.
As I said previously, you actually already know the exact size you need.